### PR TITLE
Implement OpenCV Video source

### PR DESCRIPTION
**Pull request by lixfel**
Enables the usage of typical video formats as source for ssl-vision for testing and benchmarking.

The impact of the additional OpenCV module videoio on the Bluefox SDK support has not been tested.

### CODE REVIEW

File: src/shared/capture/capture_video.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  timestamp += 1.0/capture.get(cv::CAP_PROP_FPS);
+  img.setTimeCam(timestamp);
+
+  return img;
+}
+
+void CaptureVideo::releaseFrame() {
+  //img.clear();
```
Comment (g3force) [inline src/shared/capture/capture_video.cpp:58]: Why is this commented out?

File: src/shared/capture/captureinterface.cpp [hunk 2]
```diff
@@ -46,6 +46,9 @@ void CaptureInterface::readAllParameterValues() {
 }
 
 bool CaptureInterface::copyAndConvertFrame(const RawImage & src, RawImage & target) {
+  if(target.getColorFormat() == COLOR_UNDEFINED)
```
Comment (g3force) [inline src/shared/capture/captureinterface.cpp:49]: Why was this necessary? Maybe add a short comment, explaining it.

File: src/shared/capture/capture_video.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+
+RawImage CaptureVideo::getFrame() {
+  cv::Mat frame;
+  if(!capture.read(frame)) {
+    std::cout << "End of video stream reached" << std::endl;
+    return img;
+  }
+
+  int factor = v_cap_upscale->getBool() ? 2 : 1;
```
Comment (g3force) [inline src/shared/capture/capture_video.cpp:39]: Wouldn't it make more sense to configure the scale factor, instead of a flag?

File: src/shared/capture/captureinterface.cpp (hunk 2 above)
Comment (lixfel) [reply src/shared/capture/captureinterface.cpp:49]: COLOR_UNDEFINED has a pixel size of 0, which causes a segmentation fault in the following memcpy during the first frame in capture_thread.cpp#327 (`bool bSuccess = capture->copyAndConvertFrame( pic_raw,d->video);`).

File: src/shared/capture/capture_video.cpp (hunk 3 above)
Comment (lixfel) [reply src/shared/capture/capture_video.cpp:39]: Other scalings should not be needed, this is just necessary due to the half resolution of the videos captured with the new VisionProcessor.

Comment (g3force) [reply src/shared/capture/capture_video.cpp:38]: Would be good to have some explanation of why this is necessary somewhere. I have not quite get why, yet :upside_down_face:

File: src/shared/capture/captureinterface.cpp (hunk 2 above)
Comment (g3force) [reply src/shared/capture/captureinterface.cpp]: I still do not understand why your change fixes the issue. The color format should be applied anyways, except if the image is of size zero, in which case the data is equal to `nullptr`.

File: src/shared/capture/capture_video.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    cvtColor(frame, dstImg, cv::COLOR_BGR2RGB);
+  }
+
+  timestamp += 1.0/capture.get(cv::CAP_PROP_FPS);
+  img.setTimeCam(timestamp);
+
+  return img;
+}
+
+void CaptureVideo::releaseFrame() {}
```
Comment (g3force) [inline src/shared/capture/capture_video.cpp:56]: Why not clearing the matrices here? At least add a comment with that explanation.

File: src/shared/capture/capture_video.cpp (hunk 3 above)
Comment (lixfel) [reply src/shared/capture/capture_video.cpp:38]: Typical color cameras do an upscaling during the conversion from the Bayer matrix to the color image. So 4 sensor fields (RG - GB) are interpolated into 4 RGB pixels. To reduce necessary bandwidth and improve image processing time my new vision processor does an "downsampling" (only affects the green color channel) instead (RG - GB becomes 1 RGB pixel). To use these "half" resolution videos with the same geometry and blob resolution as the regular ssl-vision is used to this upscaling option is necessary.
